### ISSUE TITLE
Question on using multiple patterns in conjunction with negation.

### ISSUE THREAD

**Issue by imackintouch** [facelessuser/wcmatch#27]
Hi Isaac. Thanks for your reply on the other thread with respect to my confusion around using multiple patterns with negation.

So here's a more precise example of what I am attempting to accomplish.

Let us say that I have a directory with the following set of files:
`foo.txt`
`foo.in`
`foo.bar`

I thought... I THOUGHT....that doing the following:
`from wcmatch import glob`
`glob.glob('!*(.in|*.bar', flags=glob.EXTGLOB|glob.NEGATE)`

would then yield:
`['foo.txt']`

Essentially, I was expecting the same behavior that one would get from the bash command line.
It appears that I may need to use another glob method from your module to achieve this?
I was having trouble getting clarity from the docs in this particular regard.

**Comment by facelessuser** [comment]
Did you mistype this pattern `'!*(.in|*.bar`? That doesn't look right.

**Comment by imackintouch** [comment]
Yes, I absolutely mistyped that. I meant:
!*(.in|*.bar)

This is what a multiple negation pattern set would look like on a bash command line.

However, I am now strongly beginning to realize that with glob.glob, I may need to have these negation patterns all separated out into a list or use glob.globlist to break such a pattern apart into discrete elements of a list.

**Comment by facelessuser** [comment]
I assume you meant to add `*.in` and `*.bar`? Remember how I mentioned NEGATE and EXTGLOB can conflict?  When I add MINUSNEGATE, it works, and when I don't, it doesn't:

```py
>>> glob.glob('!(*.in|*.bar)', flags=glob.N|glob.M|glob.E)
['foo.txt']
>>> glob.glob('!(*.in|*.bar)', flags=glob.N|glob.M)
[]
```

This is what I would expect.

**Comment by imackintouch** [comment]
Hi Isaac. What you indicated does absolutely work. What had confused me was something in the docs that read like this:
`When MINUSNEGATE is used with NEGATE [#globnegate], negate patterns are
recognized by a pattern starting with - instead of ! .`

I took that to mean that I needed to use an actual minus sign instead of an exclamation sign!

**Comment by facelessuser** [comment]
For NEGATE, that is true: `!pattern`, but you are using the EXTGLOB pattern `!(...)` which is a different thing.

Also note bash behavior:

```
$ echo !(.in|.bar)
. .. foo.bar foo.in foo.txt

$ echo !(*.in|*.bar)
foo.txt
```

**Comment by facelessuser** [comment]
I guess to clarify further.  MINUSNEGATE makes it so you use the `-` sign when doing `!pattern`. So now you use `-pattern`. EXTGLOB always uses `!(...)`. But with MINUSNEGATE you can use both together.  So while `!(*.jpg)` would find all files that are not `*.jpg`, `-!(*.jpg)` would find all files that are `*.jpg`. You are combining two different negation syntaxes.

**Comment by imackintouch** [comment]
Thanks for the clarifications on MINUSGATE usage!

With your information, I am now able to create glob.glob code examples that are behaving in the way that I want which is very cool!

I will also experiment with some of the other glob methods to see if they can be of value to me.

Once again thanks for your comments and assistance. This has been a very positive github experience for me! :-)

**Comment by facelessuser** [comment]
No problem. In general, things *should* work as documented as I did test the heck out of them, but (as you discovered earlier) there may be some areas where fatigue got the best of me and some tests didn't get the scrutiny that I intended. The more people that use the library, the more those things get caught (assuming there are still bugs left to find😏).

As it seems the question was answered, I will go ahead and close this.
